csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -319,14 +338,125 @@ public void OptimizerStep(RunOptions options)
 
         }
 
+        /// <summary>
+        /// Export a model that can be used for inferencing.
+        /// If the training session was provided with an eval model, the training session can generate
+        /// an inference model if it knows the inference graph outputs. The input inference graph outputs
+        /// are used to prune the eval model so that the inference model's outputs align with the provided outputs.
+        /// The exported model is saved at the path provided and can be used for inferencing with Ort::Session.
+        /// Note that the function re-loads the eval model from the path provided to Ort::TrainingSession
+        /// and expects that this path still be valid.
+        /// </summary>
+        /// <param name="inference_model_path">Path where the inference model should be serialized to.</param>
+        /// <param name="graphOutputNames">Names of the outputs that are needed in the inference model.</param>
+        public void ExportModelForInferencing(string inferenceModelPath, IReadOnlyCollection<string> graphOutputNames)
+        {
+            using (var cleanupList = new DisposableList<IDisposable>())
+            {
+                var outputNamesArray = ConvertNamesToUtf8(graphOutputNames, cleanupList);
+                NativeApiStatus.VerifySuccess(NativeTrainingMethods.OrtExportModelForInferencing(
+                    _nativeHandle, NativeOnnxValueHelper.GetPlatformSerializedString(inferenceModelPath),
+                    (UIntPtr)graphOutputNames.Count, outputNamesArray));
+            }
+        }
+
+        /// <summary>
+        /// Returns a contiguous buffer that holds a copy of all training state parameters
+        /// </summary>
+        /// <param name="only_trainable">Whether to only copy trainable parameters or to copy all parameters.</param>
+        public FixedBufferOnnxValue ToBuffer(bool only_trainable)
+        {
+            UIntPtr bufferSize = UIntPtr.Zero;
+            NativeApiStatus.VerifySuccess(NativeTrainingMethods.OrtGetParametersSize(_nativeHandle, out bufferSize, only_trainable));
+
+            float[] bufferMemory = new float[bufferSize.ToUInt64()];
+
+            var memInfo = OrtMemoryInfo.DefaultInstance; // CPU
+            var shape = new long[] {(long)bufferSize.ToUInt64()};
```

Comment:
nit: bufferSize.ToInt64()?

Reply:
Since `bufferSize` is defined as `UIntPtr`, `bufferSize.ToInt64()` leads to a compiler error

```
error CS1061: 'UIntPtr' does not contain a definition for 'ToInt64' and no accessible extension method 'ToInt64' accepting a first argument of type 'UIntPtr' could be found (are you missing a using directive or an assembly reference?)
```